float32 a
float32 b
float32 c
float32 d
float32 x
float32 y
float32 z
float32 orientation
time timestamp
uint32 vector_len
cloud[] vector_cloud